compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1510,10 +1518,9 @@ convertType env = go env
             pure TText
         | NameIn DA_Generics n <- t
         , n `elementOfUniqSet` metadataTys
-        , [_] <- ts =
-            pure TUnit
+        , [_] <- ts = erasedTy env
         | t == anyTyCon, [_] <- ts =
-            pure TUnit -- used for type-zonking
+            erasedTy env -- used for type-zonking
```

Comment:
Is that really what we want? Don't we only get into this situation when the type does not actually matter? Can we get here with kinds other than star?

Reply:
I thought about this for a bit and while I’m not entirely clear on this I think `Erased` does make sense for 2 reasons:

1. The motivation for `Erased` is to avoid trying to reconstruct stuff that we cannot reconstruct. If I see a `TUnit` coming from `Any` in a definition somewhere in DAML-LF, I’m pretty sure I do not want to translate this to a DAML definition that accepts `()`.
2. We never actually have values of type `Any` (there is no case in the LF conversion that would translate them to `EUnit` and that would also go horribly wrong) so an uninhabited type matches this.

I’ve added a comment.